src/main/webapp/app/exercises/shared/rating/star-rating/star-rating.component.ts

```diff
@@ -250,14 +249,12 @@ export class StarRatingComponent {
             }
 
             const newSize = this.size.match(/\d+/)![0];
-            const halfSize = (parseInt(newSize, 10) * 10) / 24;
-            const halfMargin = 0 - (parseInt(newSize, 10) * 20) / 24;
+            const halfSize = parseInt(newSize, 10) * (1 - (this.value - Math.floor(this.value)));
```

Comment:
I think we should use `Number(...)` instead of `parseInt` - and to me it is really uncelar what is happening in `(1 - (this.value - Math.floor(this.value))` and why we do this. 
Could you explain this to me? Could you introduce a method with a descriptive name to make the code easier to read and understand?

```suggestion
            const halfSize = Number(newSize) * (1 - (this.value - Math.floor(this.value)));
```

Reply:
It just calculates the size of the fractional stars. I changed it to Number and will leave it like this.